lib/axiom/fs/stream/generic_streams.js

```diff
@@ -0,0 +1,217 @@
+// Copyright 2015 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import AxiomError from 'axiom/core/error';
+import AxiomEvent from 'axiom/core/event';
+import Queue from 'axiom/fs/stream/queue';
+import ReadableStreamSource from 'axiom/fs/stream/readable_stream_source';
+import WritableStreamSource from 'axiom/fs/stream/writable_stream_source';
+
+/** @typedef function():void */
+var EventCallback;
+
+/** @typedef ReadableStream$$module$axiom$fs$stream$readable_stream */
+var ReadableStream;
+
+/** @typedef StreamsSource$$module$axiom$fs$stream$streams_source */
+var StreamsSource;
+
+/** @typedef WritableStream$$module$axiom$fs$stream$writable_stream */
+var WritableStream;
+
+/**
+ * Event value + associated callback
+ * @constructor
+ * @param {!*} value
+ * @param {!EventCallback} callback
+ */
+var EventWithCallback = function(value, callback) {
+  this.value = value;
+  this.callback = callback;
+};
+
+/**
+ * Handle a ReadableStream and a WritableStream over a generic connection (abstract).
+ *
+ * @constructor
+ * @implements {StreamsSource}
+ */
+export var GenericStreams = function() {
```

Comment:
I agree.  GenericStreams was kind of a placeholder because I couldn't think of something better.  Done